openpype/settings/lib.py

```diff
@@ -265,11 +265,43 @@ def save_project_anatomy(project_name, anatomy_data):
         raise SaveWarningExc(warnings)
 
 
+def _system_settings_backwards_compatible_conversion(studio_overrides):
+    # Backwards compatibility of tools 3.9.1 - 3.9.2 to keep
+    #   "tools" environments
+    if (
+        "tools" in studio_overrides
+        and "tool_groups" in studio_overrides["tools"]
+    ):
+        tool_groups = studio_overrides["tools"]["tool_groups"]
+        for tool_group, group_value in tool_groups.items():
+            if tool_group in METADATA_KEYS:
+                continue
+
+            variants = group_value.get("variants")
+            if not variants:
+                continue
+
+            for key in set(variants.keys()):
+                if key in METADATA_KEYS:
+                    continue
+
+                variant_value = variants[key]
+                if "environment" not in variant_value:
+                    variants[key] = {
+                        "environment": variant_value
+                    }
```

Comment:
Should this be more explicit and come with a version check? If `settings version < 3.9.2` do this?

Reply:
> Should this be more explicit and come with a version check?

It would be nice but don't know how to 100% achieve it. Since the zip files are created with "some version" which may have custom name with custom changes. It would be random until 3.9.2 realease.

But I think it would be possible to check if the version is lower than 3.9.2, so it's more clear?

Reply:
Ah of course - studios could do whatever with the version numbers. It's a bit odd that this compatible conversion logic would also potentially trigger for newer versions in the code base. Would have loved if we could've cleaned that up somehow.